### PR TITLE
Upgrade to use updated dependency-submission-toolkit

### PR DESCRIPTION
**Pull request by lseppala**
Updates to the latest dependency-submission-toolkit

Additionally:
* Adds support for transitive dependencies
* make snapshot job.correlator unique based on build target (can run multiple actions per workflow and get all results
* Misc build and testing improvements

I anticipate rework, but my goal is to get this installed in GitHub repos soon so we can start exercising our staff-ship more. If things can be addressed in future iterations, let's save it for later.

### CODE REVIEW

**Review by elireisman — APPROVED** [review]
You might want a +1 from someone better versed in Typescript but this all LGTM, nice change!

File: src/parse.test.ts
Comment (mrysav) [inline src/parse.test.ts:1]: Not going to lie, I kind of hate putting test files right alongside normal files ;) But I can let if fly...

**Review by mrysav — APPROVED** [review]

File: .github/workflows/go-action.yml
Comment (mrysav) [inline .github/workflows/go-action.yml:15]: I think this needs to be updated.

File: src/parse.test.ts
Comment (lseppala) [reply src/parse.test.ts:1]: It's takes all all types! I think I've more strongly adopted this working on a TypeScript+Go project, where unit tests in Go *must* be collocated with the source files. So apologies for ry

Since this is to be open sourced, I'm inclined to keep the tests collocated because this is the common idiom for Jest test organization (it's the example given in the [Getting Started](https://jestjs.io/docs/getting-started) guide, for instance). But I fully bless internal TS projects using other organization mehtods.

I hope we can at least agree that that `__test__/` directory (another Jest practice) is an abomination ;)

File: .github/workflows/go-action.yml
Comment (lseppala) [reply .github/workflows/go-action.yml]: _Good catch_. I knew it was too good to be true that it worked on the first run 🤦

Comment (lseppala) [reply .github/workflows/go-action.yml]: Oh! But it did work! :tada: https://github.com/dsp-testing/go-snapshot-action/runs/6836365970?check_suite_focus=true

Updating to point to `main`, which it will run once merged

**Pull request merged**
